spring-integration-core/src/main/resources/org/springframework/integration/config/xml/spring-integration-4.2.xsd

```diff
@@ -1659,6 +1659,63 @@
 		<xsd:attribute name="id" type="xsd:string" use="required" />
 	</xsd:complexType>
 
+	<xsd:element name="barrier">
+		<xsd:annotation>
+			<xsd:documentation>
+				Defines an endpoint that suspends a thread until a corresponding release message is received.
```

Comment:
I think that would be good to mention here the target class for the component.
Seems for me it should help  a bit users who migrate from the XML configuration to DSL.

According to that `Pitch Black` of questions on SO, it looks like people just don't know which classes are related to their XML config.

I don't say that we must do that for everything just in place, but at least we can start do that for new components.